### PR TITLE
Deprecate and add method for getting a thumbnail

### PR DESCRIPTION
**Pull request by sujaygarlanka**

### CODE REVIEW

File: boxsdk/object/file.py [hunk 1]
```diff
@@ -763,6 +764,31 @@ def get_thumbnail(self, extension='png', min_width=None, min_height=None, max_wi
...
 
+    @api_call
+    def get_thumbnail_representation(self, dimensions, extension='png'):
+        """
+        Retrieve a thumbnail image for the file.
+
+        :param dimensions:
+            The width by height size of this representation in pixels
+        :type dimensions:
+            `unicode`
```
Comment (swfree) [inline boxsdk/object/file.py:775]: should the dimensions example be something like `94x94`?

File: boxsdk/object/file.py [hunk 2]
```diff
@@ -763,6 +764,31 @@ def get_thumbnail(self, extension='png', min_width=None, min_height=None, max_wi
...
+        :param extension:
+            The file extension for the thumbnail, e.g. 'png' or 'jpg'
+        :type extension:
+            `unicode`
+        :returns:
+            The file contents of the thumbnail image
+        :rtype:
+            `bytes`
+        """
+        rep_hints = '[{0}?dimensions={1}]'.format(extension, dimensions)
```
Comment (swfree) [inline boxsdk/object/file.py:785]: Are dimensions required, or can you make a request for a representation without specifying dimensions?

Comment (sujaygarlanka) [reply boxsdk/object/file.py:785]: Added some documentation. Dimensions are required. https://developer.box.com/guides/representations/thumbnail-representation/

File: boxsdk/util/deprecation_decorator.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+def deprecated(message):
+    def deprecated_decorator(func):
+        def deprecated_func(*args, **kwargs):
+            warnings.simplefilter('default', DeprecationWarning)
+            warnings.warn("{} is a deprecated function. {}".format(func.__name__, message),
+                          category=DeprecationWarning,
+                          stacklevel=2)
+            return func(*args, **kwargs)
+        return deprecated_func
+    return deprecated_decorator
```
Comment (swfree) [inline boxsdk/util/deprecation_decorator.py:13]: seems like one of the travis failures is due to a missing newline here

File: test/unit/object/test_file.py [hunk 4]
```diff
@@ -817,3 +817,53 @@ def test_get_thumbnail(
...
+    mock_box_session.get.side_effect = [mock_representations_response, mock_content_response]
+
+    thumb = test_file.get_thumbnail_representation(
+        dimensions=dimensions,
+        extension=extension,
+    )
+
+    mock_box_session.get.assert_any_call(representation_url, headers={'X-Rep-Hints': '[{}?dimensions=92x92]'.format(extension)}, params={'fields': 'representations'})
+    mock_box_session.get.assert_any_call(content_url, expect_json_response=False)
+    assert thumb == mock_content_response.content
```
Comment (swfree) [inline test/unit/object/test_file.py:869]: looks like this file needs a newline as well

**Review by swfree — APPROVED** [review]

**Pull request merged**